### PR TITLE
CDATASRCS-2: Support Datasource Properties

### PR DESCRIPTION
**Pull request by ChanakaCooray**

### CODE REVIEW

File: components/org.wso2.carbon.datasource.core/src/main/java/org/wso2/carbon/datasource/rdbms/hikari/HikariConfiguration.java [hunk 1]
```diff
@@ -165,6 +168,56 @@ public boolean isAutoCommit() {
...
+        this.databaseProps = databaseProps;
+    }
+
+    /**
+     * Bean class to hold custom properties
+     */
+    @XmlRootElement(name = "property")
+    public static class DataSourceProperty {
+
+        private boolean encrypted = true;
```
Comment (kasunbg) [inline components/org.wso2.carbon.datasource.core/src/main/java/org/wso2/carbon/datasource/rdbms/hikari/HikariConfiguration.java]: This bean property doesn't seem to be used anywhere. :)

Comment (ChanakaCooray) [inline components/org.wso2.carbon.datasource.core/src/main/java/org/wso2/carbon/datasource/rdbms/hikari/HikariConfiguration.java]: Added a commit to remove unnecessary properties

**Comment by Niranjan-K** [comment]
Please create a JIRA for this and add it to your commit message..

File: components/org.wso2.carbon.datasource.core/src/main/java/org/wso2/carbon/datasource/rdbms/hikari/utils/HikariDataSourceUtils.java [hunk 2]
```diff
@@ -47,6 +48,12 @@ public static HikariConfig buildConfiguration(String xmlConfiguration) throws Da
             config.setMaximumPoolSize(configuration.getMaximumPoolSize());
             config.setMinimumIdle(configuration.getMinimumIdle());
             config.setAutoCommit(configuration.isAutoCommit());
+            if (configuration.getDatabaseProps() != null && !configuration.getDatabaseProps().isEmpty()) {
+                Properties properties = new Properties();
+                configuration.getDatabaseProps()
+                        .forEach(property -> properties.setProperty(property.getName(), property.getValue()));
```
Comment (DilanUA) [inline components/org.wso2.carbon.datasource.core/src/main/java/org/wso2/carbon/datasource/rdbms/hikari/utils/HikariDataSourceUtils.java:54]: better to move . sign to the upper part of the statement to imply continuation. But your approach could be a different standard, better check on this.

Comment (ChanakaCooray) [reply components/org.wso2.carbon.datasource.core/src/main/java/org/wso2/carbon/datasource/rdbms/hikari/utils/HikariDataSourceUtils.java:54]: Line break should have to be before the dot. That is the java 8 standard we have followed. Example[1].

[1] https://github.com/wso2/carbon-kernel/blob/master/core/src/main/java/org/wso2/carbon/kernel/internal/startupresolver/StartupOrderResolver.java

Comment (DilanUA) [reply components/org.wso2.carbon.datasource.core/src/main/java/org/wso2/carbon/datasource/rdbms/hikari/utils/HikariDataSourceUtils.java:54]: Approved...

**Review by DilanUA — COMMENTED** [review]
Approved...

**Pull request merged**